### PR TITLE
Location search using Places API + Use the same form as match creating to handle match editing

### PR DESCRIPTION
**Pull request by elkylu**
https://user-images.githubusercontent.com/37742291/126141051-64d95261-277a-4820-964b-e88efb4b6843.mp4

### CODE REVIEW

File: app/src/main/java/com/uwcs446/socialsports/services/match/MatchEntity.kt [hunk 1]
```diff
@@ -11,17 +9,11 @@ data class MatchEntity(
     val id: String = UUID.randomUUID().toString(),
     val sport: Sport = Sport.ANY,
     val title: String = "",
-    val location: LocationEntity = LocationEntity(),
+    val location: String = "",
```
Comment (SPahooja) [inline app/src/main/java/com/uwcs446/socialsports/services/match/MatchEntity.kt]: Changing the entity to string is not the best idea, we cannot query anything using the string, is it possible to change only the domain and convert it into a geohash for the entity?

File: app/src/main/java/com/uwcs446/socialsports/domain/match/Match.kt [hunk 2]
```diff
@@ -26,6 +28,8 @@ data class Match(
     fun currentPlayerCount() = teamOne.size.plus(teamTwo.size)
 }
 
-data class Location(
+@Parcelize
+data class HostLocation(
```
Comment (aaronchlam) [inline app/src/main/java/com/uwcs446/socialsports/domain/match/Match.kt]: Does the data class need to be specific to Host?

File: app/src/main/java/com/uwcs446/socialsports/services/LocationService.kt [hunk 3]
```diff
@@ -1,30 +1,23 @@
...
-
-        val geocoder = Geocoder(activity, Locale.getDefault())
-
-        return geocoder
-            .getFromLocation(
-                latLng.latitude,
-                latLng.longitude,
-                1
-            ).firstOrNull() ?: DEFAULT_ADDRESS
+    fun getPlace(placeId: String): Task<FetchPlaceResponse> {
```
Comment (aaronchlam) [inline app/src/main/java/com/uwcs446/socialsports/services/LocationService.kt]: I think you can use `suspend` here to and make use of coroutines?

File: app/src/main/java/com/uwcs446/socialsports/ui/host/details/HostDetailsFragment.kt [hunk 4]
```diff
@@ -54,8 +60,17 @@ class HostDetailsFragment : Fragment() {
         val hostGameButton: Button = binding.hostButtonHost
 
         // Location and match details of saved context
-        locationTitleTextView.text = hostDetailsViewModel.locationTitle
-        locationAddressTextView.text = hostDetailsViewModel.locationAddress
+        val placeId = hostDetailsViewModel.locationId
+        locationService
+            .getPlace(placeId)
+            .addOnSuccessListener { response: FetchPlaceResponse ->
```
Comment (aaronchlam) [inline app/src/main/java/com/uwcs446/socialsports/ui/host/details/HostDetailsFragment.kt]: I think we can use kotlin coroutines here. Here's you'd so something like
```
 viewModelScope.launch {
  try {
   val response = locationService.getPlace(placeId).await()
   ...do stuff with response
  } catch (e: Exception) {
  ...handle error
  }
}
```

File: app/build.gradle [hunk 5]
```diff
@@ -57,6 +57,8 @@ dependencies {
     implementation 'androidx.lifecycle:lifecycle-viewmodel-ktx:2.3.1'
     implementation 'androidx.navigation:navigation-fragment-ktx:2.3.5'
     implementation 'androidx.navigation:navigation-ui-ktx:2.3.5'
+    implementation 'com.android.volley:volley:1.2.0'
```
Comment (SPahooja) [inline app/build.gradle:60]: do we need to make any HTTP requests?

Comment (elkylu) [reply app/build.gradle:60]: This library is required by Google Places.

File: app/src/main/java/com/uwcs446/socialsports/domain/match/Match.kt (hunk 2 above)
Comment (elkylu) [reply app/src/main/java/com/uwcs446/socialsports/domain/match/Match.kt]: It is not specific to Host, I just wanted to distinguish between our data class and Android Location class. Can change it back to Location for sure.

**Review by aaronchlam — APPROVED** [review]

**Pull request merged**